purchase_request/models/purchase_order.py

```diff
@@ -56,11 +56,19 @@ def _purchase_request_confirm_message(self):
                     requests_dict[request_id][request_line.id] = data
             for request_id in requests_dict:
                 request = request_obj.sudo().browse(request_id)
+                mt_request_po_confirmed = self.env.ref(
+                    "purchase_request.mt_request_po_confirmed"
+                )
+                partners = request.message_follower_ids.filtered(
+                    lambda x: mt_request_po_confirmed in x.subtype_ids
+                ).mapped("partner_id")
                 message = po._purchase_request_confirm_message_content(
                     request, requests_dict[request_id]
                 )
                 request.message_post(
-                    body=message, subtype_id=self.env.ref("mail.mt_note").id
+                    body=message,
+                    subtype_id=self.env.ref("mail.mt_note").id,
```

Comment:
Why don't you put the specific subtype? And I think Odoo will handle it thus without getting partners.